### PR TITLE
Bump controller-runtime, controller-gen and controller-tools version for v1beta1

### LINKED ISSUE #2779: Bump controller runtime version to v0.10
/kind feature
/area api
/priority important-soon

**Describe the solution you'd like**

For v1beta1, bump CR release to v0.10

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api-provider-aws version: 
- Kubernetes version: (use `kubectl version`): 
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by Ankitasw**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR bumps the controller runtime version to v0.10.1 which is required for v1beta1 release

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2779

**Checklist**:
- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:
```release-note
Bump controller runtime version to v0.10.1, controller-gen to v0.7.0 and controller-tools to 0.7.0  for v1beta1 release
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -27,17 +27,17 @@ require (
 	golang.org/x/crypto v0.0.0-20210813211128-0a44fdfbc16e
 	golang.org/x/net v0.0.0-20210813160813-60bc85c4be6d
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.21.4
-	k8s.io/apiextensions-apiserver v0.21.4
-	k8s.io/apimachinery v0.21.4
+	k8s.io/api v0.22.2
+	k8s.io/apiextensions-apiserver v0.22.2
+	k8s.io/apimachinery v0.22.2
 	k8s.io/cli-runtime v0.21.4
```
[the remark this answers is not in the document]
Comment (Ankitasw) [reply go.mod:33]: Do we have anything relevant for us in the new release? If yes we can bump that as well?

**Comment by Ankitasw** [comment]
@randomvariable after changing the controller runtime version, [this](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/api/v1alpha4/awscluster_webhook_test.go#L511) Create func is changing the empty AllowedCIDRBlocks to nil

File: go.mod (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ankitasw) [reply go.mod:33]: Not sur why k8s didnt update this dependency, i ll update

**Comment by randomvariable** [comment]
let's stop on these. there's other problems in prow atm https://github.com/kubernetes/test-infra/issues/23741

**Comment by randomvariable** [comment]
Also placing a
/hold

so we can do one final v0.7 release before going to v0.8

**Comment by Ankitasw** [comment]
> Also placing a
> /hold
> 
> so we can do one final v0.7 release before going to v0.8

Do we still need to hold these? @randomvariable

**Comment by randomvariable** [comment]
Hold is staying on until the next 0.7.x release.

**Comment by randomvariable** [comment]
/unhold

Discussed this with @richardcase , and we're going to skip the v0.7.x release and go to v1.0.0

**Pull request merged**